### ISSUE TITLE
MemoTrie fails to install

### ISSUE THREAD

**Issue by joelburget** [valderman/haste-compiler#276]
Very mysterious error trying to compile MemoTrie:

```
joel@belfast ~/p/react-haskell> haste-inst install memotrie
Resolving dependencies...
Configuring MemoTrie-0.6.2...
Building MemoTrie-0.6.2...
Preprocessing library MemoTrie-0.6.2...
hastec: hastec: panic! (the 'impossible' happened)
  (GHC version 7.8.3 for x86_64-apple-darwin):
	cat_evals
<<details unavailable>>

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug

Failed to install MemoTrie-0.6.2
cabal: Error: some packages failed to install:
MemoTrie-0.6.2 failed during the building phase. The exception was:
ExitFailure 1
```

Though it compiles with just a few lines [commented out](https://github.com/joelburget/MemoTrie/commit/357c80c49d7894ac9268e7b4541f91f19b800751).

And I also tried to produce a somewhat [minimal repro](https://github.com/joelburget/MemoTrie/blob/e460bee9caf34daf5d826123d902a47c9ccc87c9/src/Data/MemoTrie.hs)

How should I go about getting more info on that panic?

**Comment by tusj** [comment]
I get the exact same error when installing binary-0.7.4.0. I tried installing with haste-inst --verbose=2,3 but no interesting info.

**Comment by worldsayshi** [comment]
I'm getting a similar error when trying to install [(my fork of) haste-perch](https://github.com/worldsayshi/haste-perch):

```bash
$ haste-cabal install ../haste-perch/
Resolving dependencies...
In order, the following will be installed:
binary-0.7.4.0 (new package)
directory-1.2.2.1 (new package)
haste-perch-0.1.0.10 +haste-inst (reinstall)
process-1.2.3.0 (new package)
Cabal-1.22.3.0 (reinstall) changes: array-0.4.0.1 added, base-4.6.0.1 added,
binary-0.7.4.0 added, bytestring-0.10.6.0 added, containers-0.4.1.0 added,
deepseq-1.4.1.1 added, directory-1.2.2.1 added, filepath-1.4.0.0 added,
pretty-1.1.3.2 added, process-1.2.3.0 added, time-1.5.0.1 added, unix-2.7.1.0
added
Warning: Note that reinstalls are always dangerous. Continuing anyway...
Notice: installing into a sandbox located at
/Users/per.fredelius/workspace/agfjord-graph-search/hstest/grammarFish/attempt/.cabal-sandbox
Configuring binary-0.7.4.0...
Configuring haste-perch-0.1.0.10...
Configuring directory-1.2.2.1...
Failed to install directory-1.2.2.1
Build log ( /Users/per.fredelius/workspace/agfjord-graph-search/hstest/grammarFish/attempt/.cabal-sandbox/logs/directory-1.2.2.1.log ):
Building haste-perch-0.1.0.10...
Building binary-0.7.4.0...
Failed to install binary-0.7.4.0
Build log ( /Users/per.fredelius/workspace/agfjord-graph-search/hstest/grammarFish/attempt/.cabal-sandbox/logs/binary-0.7.4.0.log ):
Configuring binary-0.7.4.0...
Building binary-0.7.4.0...
Preprocessing library binary-0.7.4.0...

src/Data/Binary/Builder/Base.hs:192:37: Warning:
    In the use of ‘inlinePerformIO’
    (imported from Data.ByteString.Internal):
    Deprecated: "If you think you know what you are doing, use 'unsafePerformIO'. If you are sure you know what you are doing, use 'unsafeDupablePerformIO'. If you enjoy sharing an address space with a malevolent agent of chaos, try 'accursedUnutterablePerformIO'."

src/Data/Binary/Get/Internal.hs:402:21: Warning:
    In the use of ‘B.inlinePerformIO’
    (imported from Data.ByteString.Internal):
    Deprecated: "If you think you know what you are doing, use 'unsafePerformIO'. If you are sure you know what you are doing, use 'unsafeDupablePerformIO'. If you enjoy sharing an address space with a malevolent agent of chaos, try 'accursedUnutterablePerformIO'."

src/Data/Binary/Get/Internal.hs:369:1: Warning:
    Rule "<$> to <*>" may never fire because ‘<$>’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘<$>’

src/Data/Binary/Get.hs:434:1: Warning:
    Rule "getWord8/readN" may never fire
      because ‘getWord8’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord8’

src/Data/Binary/Get.hs:435:1: Warning:
    Rule "getWord16be/readN" may never fire
      because ‘getWord16be’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord16be’

src/Data/Binary/Get.hs:436:1: Warning:
    Rule "getWord16le/readN" may never fire
      because ‘getWord16le’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord16le’

src/Data/Binary/Get.hs:437:1: Warning:
    Rule "getWord32be/readN" may never fire
      because ‘getWord32be’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord32be’

src/Data/Binary/Get.hs:438:1: Warning:
    Rule "getWord32le/readN" may never fire
      because ‘getWord32le’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord32le’

src/Data/Binary/Get.hs:439:1: Warning:
    Rule "getWord64be/readN" may never fire
      because ‘getWord64be’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord64be’

src/Data/Binary/Get.hs:440:1: Warning:
    Rule "getWord64le/readN" may never fire
      because ‘getWord64le’ might inline first
    Probable fix: add an INLINE[n] or NOINLINE[n] pragma on ‘getWord64le’
hastec: hastec: panic! (the 'impossible' happened)
  (GHC version 7.8.3 for x86_64-apple-darwin):
	cat_evals
<<details unavailable>>

Please report this as a GHC bug:  http://www.haskell.org/ghc/reportabug

Installed haste-perch-0.1.0.10
cabal: Error: some packages failed to install:
Cabal-1.22.3.0 depends on binary-0.7.4.0 which failed to install.
binary-0.7.4.0 failed during the building phase. The exception was:
ExitFailure 1
directory-1.2.2.1 failed during the configure step. The exception was:
user error
('/Users/per.fredelius/Library/Haskell/ghc-7.8.3/lib/haste-compiler-0.5/bin/hastec'
exited with an error:
hastec: <command line>: cannot satisfy -package-id
Cabal-1.22.3.0-9534d9426bbb252779e68b3ba59a8553:
Cabal-1.22.3.0-9534d9426bbb252779e68b3ba59a8553 is unusable due to missing or
recursive dependencies:
array-0.5.0.0-98aa445e59f3eb0c886795ff07406d84
base-4.7.0.1-df210ede1eb79477fef5662549c32927
binary-0.7.1.0-a8503b07e0b6021b0f8b0ea5381b25a6
bytestring-0.10.4.0-4aa78c8ca7b6b65993eefc131f7d94fa
containers-0.5.5.1-0d8db9193d3e3371e0142bcc8a4a0721
deepseq-1.3.0.2-8f63133c1b77f3b3190f04893cf340e4
directory-1.2.1.0-af5afa2b9b551d3fec1a32d6bfd8bd72
filepath-1.3.0.2-1580a61d3226e4be45fe2130dc2881e3
pretty-1.1.1.1-055fb3b6b88c48ad1bf91e5cf35828d1
process-1.2.0.0-3a472e9c66165e506513ea8f145681a0
time-1.4.2-d6766dce59812a4b19375d9595549a8b
unix-2.7.0.1-8adde3f1f2079286da56b30898c2d703
(use -v for more information)
)
process-1.2.3.0 depends on directory-1.2.2.1 which failed to install.
```

**Comment by worldsayshi** [comment]
It happened after I did a clean slate procedure and pulled and installed the latest version of haste from the repo. I did that in while trying to get away from other various haste-cabal issues. Not sure how clean slate the clean slate procedure actually is. Lot's of things can have caused the current state of the environment.

**Comment by valderman** [comment]
Fixed on master with GHC 7.10.
